geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/ConvertToGeoFile.scala

```diff
@@ -0,0 +1,238 @@
+      val format = ExportFormat(context.getProperty(OutputFormat).getValue).get
+      val gzip = Option(context.getProperty(GzipLevel).evaluateAttributeExpressions().getValue).map(_.toInt)
+      val headers = Option(context.getProperty(IncludeHeaders).getValue).forall(_.toBoolean)
+
+      output = session.create(input)
+
+      var result: IngestResult = null // have to update the attributes after closing the output file
+
+      def export(exporter: FeatureExporter): Unit = {
+        try {
+          exporter.start(sft)
+          val callback = new ConverterCallback() {
+            override def apply(features: Iterator[SimpleFeature]): Long = {
+              exporter.export(features)
+              0L
+            }
+          }
+          result = convert(context, session, input, sft, callback)
+        } finally {
+          CloseWithLogging(exporter)
+        }
+      }
+
+      if (format.streaming) {
+        output = session.write(output, new OutputStreamCallback() {
+          override def process(out: OutputStream): Unit = {
+            // note: avro handles compression internally
+            lazy val stream = {
+              val compressed = gzip match {
+                case None    => out
+                case Some(c) => new GZIPOutputStream(out) { `def`.setLevel(c) } // hack to access the protected deflate level
+              }
+              new FlowFileExportStream(compressed)
+            }
+            val exporter = format match {
+              case ExportFormat.Arrow   => new ArrowExporter(stream, ConvertToGeoFile.getArrowHints(sft), Map.empty)
+              case ExportFormat.Avro    => new AvroExporter(new FlowFileExportStream(out), gzip)
+              case ExportFormat.Bin     => new BinExporter(stream, new Hints())
+              case ExportFormat.Csv     => DelimitedExporter.csv(stream, headers)
+              case ExportFormat.Gml2    => GmlExporter.gml2(stream)
+              case ExportFormat.Gml3    => GmlExporter(stream)
+              case ExportFormat.Json    => new GeoJsonExporter(stream)
+              case ExportFormat.Leaflet => new LeafletMapExporter(stream)
+              case ExportFormat.Tsv     => DelimitedExporter.tsv(stream, headers)
+              // shouldn't happen unless someone adds a new format and doesn't implement it here
+              case _ => throw new NotImplementedError(s"Export for '$format' is not implemented")
+            }
+            export(exporter)
+          }
+        })
+      } else {
+        val dir = Files.createTempDirectory("gm-nifi")
+        try {
+          val file = new File(dir.toFile, s"export.${format.extensions.head}")
+          val exporter = format match {
+            case ExportFormat.Orc     => new OrcFileSystemExporter(file.getAbsolutePath)
+            case ExportFormat.Parquet => new ParquetFileSystemExporter(file.getAbsolutePath)
+            // case ExportFormat.Shp     => new ShapefileExporter(file) // TODO zip up dir??
```

Comment:
Should we remove this comment and add a ticket?  Or implement?

Reply:
i wasn't sure anyone would ever want this, and it was more complicated to implement since it creates multiple files... my thinking was to just leave it as a TODO in case anyone ever comes looking, since I'd rather not open a ticket we don't plan on implementing.

Reply:
Fair.

I could see shapefile support as a way to update contextual layers.  For example, one could have a process that writes a new shapefile to S3 as the result of some process.  

I'm still on the fence about whether or not that's a great idea or not.